### DISCUSSION TITLE
Code-formatting rules using clang-format-11 and gofmt tools applied to Certifier Framework code base.

### DISCUSSION THREAD

**Discussion by gapisback** [ccc-certifier-framework/certifier-framework-for-confidential-computing#185] [Announcements]
Hello, team --

A few days ago, under commit @SHA [c1f59c6](https://github.com/vmware-research/certifier-framework-for-confidential-computing/commit/c1f59c6e35eb4c891252ae63b6dffb25d9e7401c), a collection of clang-formatting rules have been applied to the source code and integrated to `/main`.

PR is: https://github.com/vmware-research/certifier-framework-for-confidential-computing/pull/172

Relevant dev workflow documents will be updated shortly, but in the interim, here are the things you should be aware of while doing code development in this repo:

- We have tried to stay close[r] to the existing code base, which is based on Google’s C++ coding style. We have incorporated a set of rules in a [.clang-format](https://github.com/vmware-research/certifier-framework-for-confidential-computing/blob/main/.clang-format) file that will now live at the root of the Certifier repo.

- This new source formatting requirement is checked by a new script, [CI/scripts/check-srcfmt.sh](https://github.com/vmware-research/certifier-framework-for-confidential-computing/tree/main/CI/scripts#:~:text=2%20months%20ago-,check%2Dsrcfmt.sh,-(File))

- We have standardized on `clang-format-11` version of the tool
- Any source-code formatting errors will be caught by this script and your CI build-job will fail in very early stage. So, it’s better if you check your formatting before pushing to GitHub to trigger CI builds.

- In your active dev branches, you can run this to fix all formatting errors : `$ CI/scripts/check-srcfmt.sh fixall`

- The code-formatting for a small collection of Go code in the repository is verified by use of the `gofmt` tool, and implemented in CI by the [CI/scripts/check-gofmt.sh](https://github.com/vmware-research/certifier-framework-for-confidential-computing/blob/main/CI/scripts/check-gofmt.sh) script. This change was implemented under PR #125 .

All developers are requested to be aware of these code-formatting rules and to apply the relevant scripts / tools prior to submitting code for PR-reviews.

Thanks, in advance.

**Comment by jinbpark** [comment]
Nice description! One suggestion. How about adding a formatting tool for shell programs? For example, ISLET is using `shfmt`.

**Reply by gapisback** [reply]
yes, @jinbpark -- Agreed. We do use `shellcheck` and `shfmt` while developing these Bash scripts, but those checks have not been included in CI workflows.

Why don't you log a new issue requesting this addition, so it can be included in our development tasks? Thanks!

**Reply by jinbpark** [reply]
https://github.com/vmware-research/certifier-framework-for-confidential-computing/issues/187  --> Created.
